lib/cockpit/cockpit-auth-miq

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+require "net/http"
+require "json"
+require "uri"
+require 'socket'
+
+MAX_AUTH_SIZE = 64 * 1024
+AUTH_FD = 10
+
+host = ARGV[-1]
+success=false
+url = URI.parse('http://localhost:3000/api/')
+if ENV['MQ_API_PATH']
+    url = URI.parse(ENV['MQ_API_PATH'])
+end
+
+ios = IO.for_fd(AUTH_FD)
+token = ios.readpartial(MAX_AUTH_SIZE)
+
+req = Net::HTTP::Get.new(url)
+req['X-Auth-Token'] = token
+res = Net::HTTP.start(url.hostname, url.port) {|http|
+  http.request(req)
+}
+
+json_resp = { 'error' => 'Unexpected Error' }
+
+if res.code == "200"
+    begin
+        success = true
+    rescue
```

Comment:
This `begin ... rescue` block will never get to rescue because its around `success = true` which will never fail.